### ISSUE TITLE
Cover Block: Incorrect display of color-contrast warning

### ISSUE THREAD

**Issue by kathrynwp** [WordPress/gutenberg#52800]
### Description

In WP 6.3-RC1, the Cover block incorrectly displays a color-contrast warning, where it did not in 6.2.2.

### Step-by-step reproduction instructions

1. Activate Twenty Twenty-Three
2. Add a Cover block to a page or post
3. Add a dark background image to the block (without modifying any opacity or other settings)
4. Add some text (without modifying the default colours)

### Result

This message is displayed:

> This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.

### Expected

No color-contrast warning should be displayed if there are no contrast issues.

### 6.3-RC1:

<img width="1509" alt="Add_New_Page_‹_LearnWP_—_WordPress" src="https://github.com/WordPress/gutenberg/assets/7662449/91e42aa3-e90b-4819-b2cb-f2162c412b88">

### Compare to 6.2.2:

<img width="1281" alt="Add_New_Page_‹_Testing_Testing_123_—_WordPress_and_Messenger" src="https://github.com/WordPress/gutenberg/assets/7662449/2f971491-c642-4140-949a-c0edc2fbc959">

### Environment info

- WP 6.3-RC1
- Twenty Twenty-Three 1.1
- No plugins active except WP Beta Tester

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by kathrynwp** [comment]
In addition, I noticed that changing the colour of the Paragraph block within the Cover block to a colour that's actually not contrasty enough **removes** the contrast warning. 

In other words:

The contrast warning is present when it shouldn't be, and absent when it should be there.

<img width="1082" alt="Edit_Page_“Cover_Block”_‹_LearnWP_—_WordPress" src="https://github.com/WordPress/gutenberg/assets/7662449/96cf574e-e8ca-4217-9acb-91db852f5e4c">

<img width="1154" alt="Edit_Page_“Cover_Block”_‹_LearnWP_—_WordPress" src="https://github.com/WordPress/gutenberg/assets/7662449/ac3908e6-65c2-486e-989a-da11af7b26f6">

**Comment by richtabor** [comment]
Thanks @kathrynwp — this is a duplicate of https://github.com/WordPress/gutenberg/issues/50531 so I'll close this one. Please feel free to add additional feedback to that issue if needed. 🙇‍♂️
